service/src/ConstConfigIOGroup.hpp

```diff
@@ -0,0 +1,174 @@
+            /// @return  a function that should be used when formatting the given signal.
+            ///
+            /// @see geopm::Agg
+            std::function<std::string(double)> format_function(const std::string &signal_name) const override;
+            /// @return a string description for signal_name, if defined.
+            std::string signal_description(const std::string &signal_name) const override;
+            /// @throws geopm::Exception there are no controls supported by the ConstConfigIOGroup
+            std::string control_description(const std::string &control_name) const override;
+            /// @return M_SIGNAL_BEHAVIOR_CONSTANT 
+            int signal_behavior(const std::string &signal_name) const override;
+            /// @brief Does nothing; this IOGroup does not provide any controls.
+            ///
+            /// @param save_path this argument is ignored
+            void save_control(const std::string &save_path) override;
+            /// @brief Does nothing; this IOGroup does not provide any controls.
+            ///
+            /// @param save_path this argument is ignored
+            void restore_control(const std::string &save_path) override;
+            /// @return the name of the IOGroup
+            std::string name(void) const override;
+            /// @return the name of the plugin to use when this plugin is
+            ///         registered with the IOGroup factory
+            ///
+            /// @see geopm::PluginFactory
+            static std::string plugin_name(void);
+            /// @return a pointer to a new ConstConfigIOGroup object
+            ///
+            /// @see geopm::PluginFactory
+            static std::unique_ptr<IOGroup> make_plugin(void);
+
+        private:
+            struct m_signal_desc_s {
+                json11::Json::Type type;
+                bool required;
+            };
+
+            struct m_signal_info_s {
+                m_signal_info_s(
+                    int units,
+                    int domain,
+                    std::function<double(const std::vector<double> &)> agg,
+                    std::string desc,
+                    std::vector<double> values):
+
+                    units(units),
+                    domain(domain),
+                    agg_function(agg),
+                    description(desc),
+                    values(values)
+                {}
+
+                int units;
+                int domain;
+                std::function<double(const std::vector<double> &)> agg_function;
+                std::string description;
+                std::vector<double> values;
+            };
+
+            struct m_signal_ref_s {
+                std::shared_ptr<m_signal_info_s> signal_info;
+                int domain_idx;
+            };
+
+            void parse_config_json(const std::string &config);
+            static json11::Json construct_config_json_obj(
+                const std::string &config);
```

Comment:
If you've going to have lines earlier in the file that go beyond column ~75, please keep that style consistent throughout the file.  I'd still prefer to see a line break if you get up to column 100 or so.

```suggestion
            static json11::Json construct_config_json_obj(const std::string &config);
```

Reply:
Yeah, let me see if I can clean it up for consistency.